### PR TITLE
Sway: module

### PR DESCRIPTION
**Pull request by ndowens**
###### Motivation for this change

Adds service module for the swaywm 

###### Things done

- [x ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x ] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ n/a] Tested execution of all binary files (usually in `./result/bin/`)
- [x ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by ndowens** [comment]
To anyone that reviews this, I was unsure if I should add or not on the import removemodule part, since it is based on i3, my guess probably no need to remove i3 when someone chooses sway, but added it just in case and will remove if need to

File: nixos/modules/module-list.nix [hunk 1]
```diff
@@ -542,6 +542,7 @@
   ./services/web-servers/apache-httpd/default.nix
   ./services/web-servers/caddy.nix
   ./services/web-servers/fcgiwrap.nix
+  ./services/web-servers/hiawatha.nix
```
Comment (fpletz) [inline nixos/modules/module-list.nix]: This line shouldn't be part of this PR,

File: nixos/modules/services/x11/window-managers/sway.nix [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+  configFile = mkOption {
+    default = null;
+    type = with types; nullOr path;
+    description = ''
+                  Location of the Sway configuration file.
+		  '';
+      };
+  };
+};
```
Comment (fpletz) [inline nixos/modules/services/x11/window-managers/sway.nix]: Please fix the indention of the options.

File: nixos/modules/services/x11/window-managers/sway.nix [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    name = "sway";
+    start = ''
+            ${pkgs.sway}/bin/sway ${optionalString (cfg.configFile != null)
+	      "-c \"${cfg.configFile}\""
+	      } &
+	      waitPID=$!
+	     '';
+	     }];
+	     environment.systemPackages = [ pkgs.sway ];
+	};
```
Comment (fpletz) [inline nixos/modules/services/x11/window-managers/sway.nix]: Indention is also pretty wild here.

**Review by fpletz — CHANGES_REQUESTED** [review]

File: nixos/modules/module-list.nix (hunk 1 above)
Comment (ndowens) [inline nixos/modules/module-list.nix]: Weird that line is there. I branched off to separate 😀 Will do when I get home

**Comment by ndowens** [comment]
@fpletz gonna close this and create another PR, tried updating the commit but it didnt seem to do anygood
